electron/launcher.ts

```diff
@@ -0,0 +1,292 @@
+  launchArguments?: string,
+  runner: Runner = 'legendary'
+) {
+  const isLegendary = runner == 'legendary'
+  const isGOG = runner == 'gog'
+  //   const isExternal = runner == 'heroic'
+  const epicOffline = isLegendary && (await isEpicServiceOffline())
+  const isOffline = isLegendary && (!(await isOnline()) || epicOffline)
+  let envVars = ''
+  let gameMode: string
+  const gameSettings =
+    GameConfig.get(appName).config ||
+    (await GameConfig.get(appName).getSettings())
+
+  const {
+    winePrefix,
+    wineVersion,
+    wineCrossoverBottle,
+    otherOptions,
+    useGameMode,
+    showFps,
+    nvidiaPrime,
+    launcherArgs = '',
+    showMangohud,
+    audioFix,
+    autoInstallDxvk,
+    offlineMode,
+    enableFSR,
+    maxSharpness,
+    enableResizableBar,
+    enableEsync,
+    enableFsync,
+    targetExe
+  } = gameSettings
+
+  const { discordRPC } = await GlobalConfig.get().getSettings()
+  const DiscordRPC = discordRPC ? makeClient('852942976564723722') : null
+  const runOffline = isOffline || offlineMode ? '--offline' : ''
+  const exe = targetExe && isLegendary ? `--override-exe ${targetExe}` : ''
+  const gameInfo = await getGameInfo(appName, runner)
+  const isMacNative = gameInfo.is_mac_native
+  // const isLinuxNative = gameInfo.is_linux_native
+  const mangohud = showMangohud ? 'mangohud --dlsym' : ''
+  let runWithGameMode = ''
+  if (discordRPC) {
+    // Show DiscordRPC
+    // This seems to run when a game is updated, even though the game doesn't start after updating.
+    let os: string
+
+    switch (process.platform) {
+      case 'linux':
+        os = 'Linux'
+        break
+      case 'win32':
+        os = 'Windows'
+        break
+      case 'darwin':
+        os = 'macOS'
+        break
+      default:
+        os = 'Unknown OS'
+        break
+    }
+
+    logInfo('Updating Discord Rich Presence information...', LogPrefix.Backend)
+    DiscordRPC.updatePresence({
+      details: gameInfo.title,
+      instance: true,
+      largeImageKey: 'icon',
+      large_text: gameInfo.title,
+      startTimestamp: Date.now(),
+      state: 'via Heroic on ' + os
+    })
+  }
+  if (isLinux) {
+    // check if Gamemode is installed
+    await execAsync(`which gamemoderun`)
+      .then(({ stdout }) => (gameMode = stdout.split('\n')[0]))
+      .catch(() => logWarning('GameMode not installed', LogPrefix.Backend))
+
+    runWithGameMode = useGameMode && gameMode ? gameMode : ''
+  }
+  if (
+    isWindows ||
+    (isMac && isMacNative) ||
+    (isLinux && gameInfo.install.platform == 'linux')
+  ) {
+    let command = ''
+    if (runner == 'legendary') {
+      command = `${legendaryBin} launch ${appName} ${exe} ${runOffline} ${
+        launchArguments ?? ''
+      } ${launcherArgs}`
+      logInfo(['Launch Command:', command], LogPrefix.Legendary)
+    } else if (runner == 'gog') {
+      // MangoHud,Gamemode, nvidia prime can be used in native titles
+      command = `${mangohud} ${runWithGameMode} ${
```

Comment:
Although I believe this is a valid approach, the one we are using below for Epic is more organized since it puts all arguments inside a variable on the command.
It is also not perfect and can be improved. But we can think about it on another PR.